atomicapp/requirements.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+from atomicapp.constants import REQUIREMENT_FUNCTIONS
+from atomicapp.plugin import Plugin
+
+plugin = Plugin()
+plugin.load_plugins()
+
+logger = logging.getLogger(__name__)
+
+
+class Requirements:
+
+    """
+    Requirements will search what is currently being used under
+    the requirements section of Nulecule and deploy it in according
+    to the graph variables as well as whether or not said requirement
+    exists within the provider.
+
+    The REQUIREMENTS_FUNCTIONS dictionary maps all current requirement
+    names to the function names for each provider.
+
+    For example, the persistentVolume requirement in Nulecule is mapped as
+    the persistent_storage function within each provider.
+
+    Requirements tries to be as modular as possible.
+    """
+
+    def __init__(self, config, basepath, graph, provider, dryrun):
+        self.config = config
+        self.basepath = basepath
+        self.graph = graph
+        self.dryrun = dryrun
+
+        # We initialize the provider in order to gather essential information such as kubectl config
+        self.provider = plugin.getProvider(provider)(config, basepath, dryrun)
```

Comment:
Would rather break this up so it isn't two chained function calls. In nulecule/lib.py we do it like this:

```
provider_class = self.plugin.getProvider(provider_key)
provider_class(self.get_context(), self.basepath, dry)
```